packages/server/graphql/queries/helpers/removeStalePrevUsedRepoIntegations.ts

```diff
@@ -0,0 +1,15 @@
+import ms from 'ms'
+import getRedis from '../../../utils/getRedis'
+import {RemoteRepoIntegration} from './fetchAllRepoIntegrations'
+
+const removeStalePrevUsedRepoIntegrations = async (
+  prevUsedRepoIntegrations: RemoteRepoIntegration[] | null,
+  prevUsedRepoIntegrationsKey: string
+) => {
+  if (!prevUsedRepoIntegrations) return
+  const sixMonthsAgo = Date.now() - ms('180d')
+  const redis = getRedis()
+  await redis.zremrangebyscore(prevUsedRepoIntegrationsKey, '-inf', sixMonthsAgo)
```

Comment:
-2 does the previously used repos have a TTL? let's say a user stops using us for a month or a year. how will that key get removed from redis? if redis died, how would we recover the previous used repos?
for persisted data, postgres works really well because we keep that on disk where it's cheap & we have replicas, perform backups, etc. 
we have to treat redis like its data is ephemeral & could die at any moment without a loss in UX.

if we stick with redis, perhaps we have a function that calculates the previously used repos by polling the tasks on that team if the value in redis isn't found? then we can put a TTL on it

Reply:
We don't have a TTL as it's a sorted set using the timestamp as the score. We could use a key-value pair with a TTL but updating that cache would be more complex. Currently, the stale values are removed when anyone in the team reads repoIntegrations. If a team stops using us for a year, the values wouldn't be removed. How much of a concern is that?

> we have to treat redis like its data is ephemeral & could die at any moment without a loss in UX

For sure, if Redis died, there wouldn't be any real loss to the UX. repoIntegrations checks the prevUsed cache and allRepos cache. If allRepos cache is empty, it'll reach out to the network. If prevUsed cache is empty, it'll show the allRepos cache or allRepos results from the network

Reply:
>  How much of a concern is that?

if the values never expire, we'll want to move this to postgres. redis is just for ephemeral data-- think of it like an really big in-memory javascript object. if it grows unbounded, then we'd call that a memory leak.

> wouldn't be any real loss to the UX.

wouldn't we lose the previously used repos? for example, on the product team say we have 2 previous used repos: parabol & action-chronos. if redis died, the user would now see all the repos instead of the 2 that they've been using. as a user, would that feel odd?

Reply:
I've pushed an update that expires the prevRepos sorted set after 6 months of inactivity 👍 

To recap our chat yesterday, we felt that using Redis was the right solution because if Redis were to go down, we would fetch all repos from the network. The user wouldn't see their prevUsed repos at the top of the list, but they would be able to easily search for them